test/e2e/toolchaincluster_test.go

```diff
@@ -55,6 +61,26 @@ func verifyToolchainCluster(t *testing.T, await *wait.Awaitility, otherAwait *wa
 		util.ValidateKubeClient(t, targetClient, otherAwait.Namespace, &toolchainv1alpha1.ToolchainClusterList{})
 	})
 
+	t.Run("status is populated with info from kubeconfig", func(t *testing.T) {
+		if !tcStatusShouldBePopulated {
+			t.Skip("not applicable for this cluster type")
+			return
+		}
+		secret := corev1.Secret{}
+		require.NoError(t, await.Client.Get(context.TODO(), client.ObjectKey{Name: current.Spec.SecretRef.Name, Namespace: current.Namespace}, &secret))
+		apiConfig, err := clientcmd.Load(secret.Data["kubeconfig"])
```

Comment:
should we ensure that kubeconfig key exists/is not empty in the secret data or isn't necessary?